pages/api/messages.js

```diff
@@ -9,45 +22,102 @@ const allowedFields = [
 ];
 
 export default function handler(req, res) {
-  const { fields, fromUserId } = req.query;
-
-  // <--- this part filters by field ---->
-  let filteredThreads = messagesData.map((thread) => {
-    // add last message to response, to make accessing the datetime of the last message and summary easier
-    thread.last_message = thread.messages[thread.messages.length - 1];
-    return thread;
-  });
-
-  if (fields) {
-    const filteredFields = fields
-      .split(',')
-      .filter((field) => allowedFields.includes(field));
-
-    if (filteredFields.length > 0) {
-      filteredThreads = messagesData.map((thread) => {
-        const filteredThread = {};
-
-        filteredFields.forEach((key) => {
-          filteredThread[key] = thread[key];
-        });
+  if (req.method === 'POST') {
+    const { action } = req.query;
+    const { index, fromUserId, like, text } = req.body;
 
-        return filteredThread;
-      });
+    if (!fromUserId) {
+      return res
+        .status(400)
+        .send('Bad request. fromUserId needs to be provided');
+    }
+    if (typeof index !== 'number') {
+      return res.status(400).send('Bad request. index needs to be a number');
```

Comment:
This is causing an issue with `submitMessage` action, it doesn't have an index. Move the check to where index is used.